src/transformers/models/swin/configuration_swin.py

```diff
@@ -0,0 +1,133 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Swin Transformer model configuration"""
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+SWIN_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "microsoft/swin-tiny-patch4-window7-224": "https://huggingface.co/microsoft/swin-tiny-patch4-window7-224/resolve/main/config.json",
+    # See all Swin models at https://huggingface.co/models?filter=swin
+}
+
+
+class SwinConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`SwinModel`]. It is used to instantiate an Swin
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the Swin
+    [microsoft/swin-tiny-patch4-window7-224](https://huggingface.co/microsoft/swin-tiny-patch4-window7-224)
+    architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        image_size (`int`, *optional*, defaults to 224):
+            Size (resolution) of the input images.
+        patch_size (`int`, *optional*, defaults to 4):
+            Size of patches.
+        num_channels (`int`, *optional*, defaults to 3):
+            Number of channels in input images.
+        embed_dim (`int`, *optional*, defaults to 96):
+            Dimensionality of patch embedding.
+        depths (`list(int)`, *optional*, defaults to [2, 2, 6, 2]):
+            Depth of each layer in the Transformer encoder.
+        num_heads (`list(int)`, *optional*, defaults to [3, 6, 12, 24]):
+            Number of attention heads in each layer of the Transformer encoder.
+        window_size (`int`, *optional*, defaults to 7):
+            Size of windows.
+        mlp_ratio (`float`, *optional*, defaults to 4.0):
+            Ratio of MLP hidden dimesionality to embedding dimensionality.
```

Comment:
```suggestion
            Ratio of MLP hidden dimensionality to embedding dimensionality.
```
This is something that is used in a lot of timm models, as well as the original implementation of Swin Transformer.

Note that for most existing models in the library, we don't use `mlp_ratio`, but we rather define an `intermediate_size`, which determines the dimensionality of the intermediate MLPs (e.g., ViT's config has an `intermediate_size` attribute set to 3072).

This is a bit of an inconsistency. Are we ok we proceeding with `mlp_ratio`?

cc @LysandreJik @sgugger

Reply:
Update: I see that I've also used `mlp_ratios` in SegFormer (which consists of several blocks).

Reply:
Might be better to use `intermediate_size` indeed, for consistency with the other architectures.

Reply:
Would it be possible to make an exception for Swin? The dimensionality of the intermediate MLPs is not constant throughout all the layers in the model.

Reply:
Ah in that case a ratio makes more sense. Thanks for explaining!

Reply:
The suggestion has not been added and the typo is still there.